presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -124,7 +124,7 @@ public Operator createOperator(DriverContext driverContext)
 
         private ConnectorPageSink createPageSink()
         {
-            boolean connectorCommitRequired = connectorCommitStrategy == ConnectorCommitStrategy.LIFESPAN_COMMIT;
+            boolean connectorCommitRequired = connectorCommitStrategy == ConnectorCommitStrategy.LIFESPAN_COMMIT || connectorCommitStrategy == ConnectorCommitStrategy.TASK_COMMIT;
```

Comment:
nit: How about make `connectorCommitRequired` a enum property?

Reply:
@shixuan-fan : I see. You mean similar to https://github.com/prestodb/presto/blob/3e4f18232ef791188f3388377fc0889922f23333/presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java#L58-L60 ?  That's a great idea!

Reply:
Done